pkg/sql/tablemetadatacache/table_metadata_updater.go

```diff
@@ -82,30 +92,31 @@ func (u *tableMetadataUpdater) updateCache(ctx context.Context) (updated int, er
 	// upsertQuery is the query used to upsert table metadata rows,
 	// it is reused for each batch to avoid allocations between batches.
 	upsert := newTableMetadataBatchUpsertQuery(tableBatchSize)
-	it := newTableMetadataBatchIterator(u.ie, u.testKnobs.GetAOSTClause())
+	it := newTableMetadataBatchIterator(u.ie, u.spanStatsFetcher, u.testKnobs.GetAOSTClause())
 	estimatedRowsToUpdate, err := u.getRowsToUpdateCount(ctx)
 	if err != nil {
 		log.Errorf(ctx, "failed to get estimated row count. err=%s", err.Error())
 	}
 	estimatedBatches := int(math.Ceil(float64(estimatedRowsToUpdate) / float64(tableBatchSize)))
 	batchNum := 0
 	for {
-		more, err := it.fetchNextBatch(ctx, tableBatchSize)
+		more, batchErr := it.fetchNextBatch(ctx, tableBatchSize)
 		if err != nil {
```

Comment:
Should this change to:
```suggestion
		if batchErr != nil {
```

Reply:
Ah yes, good catch. Done. Also added test coverage for this conditional section.